### PR TITLE
emit warning when pytest.warns receives unknown keyword arguments

### PR DESCRIPTION
**Pull request by nicoddemus**

### CODE REVIEW

**Comment by nicoddemus** [comment]
Not sure, this feels like the situation where people will be glad their stuff breaks, even on a minor release. Well, let me know if you really think this should go into features and I will gladly rebase it. 😁  👍

**Comment by asottile** [comment]
yeah I mean _I'd_ be happy with it in `master`, but I can imagine the "you broke my code in a bugfix release!?!?!" people being unhappy

up to you, I'm ok with it either way

**Comment by RonnyPfannschmidt** [comment]
thanks for the outline, can we make this a warning on the features branch instead - its clearly wrong behavior, but also clearly a api contract we accidentally made
so the clan way out is starting with a warning, and going for a real error the next major release

**Comment by nicoddemus** [comment]
Hmmm. OK. I will do that later then.

File: testing/deprecated_test.py [hunk 1]
```diff
@@ -238,3 +239,11 @@ def test_python_deprecation(testdir):
         )
     else:
         assert msg not in result.stdout.str()
+
+
+def test_pytest_warns_unknown_kwargs():
+    with pytest.warns(
```
Comment (nicoddemus) [inline testing/deprecated_test.py]: inception

**Pull request merged**